rust/private/rustc.bzl

```diff
@@ -693,6 +693,9 @@ def establish_cc_info(ctx, crate_info, toolchain, cc_toolchain, feature_configur
         if CcInfo in dep:
             cc_infos.append(dep[CcInfo])
 
+    if crate_info.type in ("rlib", "lib") and toolchain.extra_libstd_and_allocator_ccinfo:
```

Comment:
Hmm maybe there's a potential optimization, if we depend on an rlib rust_library (a target that provides CrateInfo and CcInfo at the same time), we can assume that it will depend on the allocator ccinfo. If there are no such deps, the current target has to depend on the CcInfo.

Either dismiss on add a todo, let's not implement it in this PR.

Reply:
added todo